### PR TITLE
Ruby3

### PR DESCRIPTION
**Pull request by jreidinger**
PReparation for ruby3 and fix wrong fix for handling iso schema.

### CODE REVIEW

File: src/lib/y2packager/clients/inst_productsources.rb [hunk 1]
```diff
@@ -354,7 +354,8 @@ def NormalizeURL(url_string)
       return url_string if url_string.nil? || url_string == ""
 
       # unescape it
-      url_string = URI.unescape(url_string)
+      url_string = CGI.unescape(url_string)
+      log.info "unescaped url_string #{url_string.inspect}"
```
Comment (lslezak) [inline src/lib/y2packager/clients/inst_productsources.rb:358]: URLs might potentially contain sensitive data like password, URLs should be always filtered using the `Yast::URL.HidePassword(url)` call when logging them.

File: src/lib/y2packager/self_update_addon_repo.rb [hunk 2]
```diff
@@ -88,8 +88,9 @@ def self.present?(path = REPO_PATH)
     # @return [Boolean] true on success, false if failed
     #
     def self.create_repo(path = REPO_PATH)
-      ret = Yast::Pkg.SourceCreateType("dir://#{URI.escape(path)}?alias=SelfUpdate0",
-        "", "Plaindir")
+      ret = Yast::Pkg.SourceCreateType(
+        "dir://#{Yast::URLRecode.EscapePath(path)}?alias=SelfUpdate0", "", "Plaindir"
```
Comment (lslezak) [inline src/lib/y2packager/self_update_addon_repo.rb:92]: What's the difference to `path.split("/").map { |d| CGI.escape(d) }.join("/")` used below?

**Review by lslezak — COMMENTED** [review]
Password logging

File: src/lib/y2packager/self_update_addon_repo.rb (hunk 2 above)
Comment (jreidinger) [reply src/lib/y2packager/self_update_addon_repo.rb:92]: well, this is dedicated method just for paths. Difference to below method. To be honest I am not sure which one is better as both mangle if it gets non path part of URL and encode it wrongly.

Comment (jreidinger) [reply src/lib/y2packager/self_update_addon_repo.rb:92]: also that split is used in test code where I do not want too much rely on other code. But here in "production" code I would like to delete escaping.

**Review by lslezak — APPROVED** [review]

**Pull request merged**